src/service/viewer-impl.js

```diff
@@ -445,7 +452,40 @@ export class Viewer {
    * @return {!VisibilityState}
    */
   getVisibilityState() {
-    return this.visibilityState_;
+    // The viewer has informed us that the doc is no longer active, meaning
+    // the viewer closed or it swiped to another doc.
+    if (this.visibilityState_ === VisibilityState.UNLOADED ||
+        this.visibilityState_ === VisibilityState.HIDDEN) {
```

Comment:
This will look awesome in the `setVisibilityState_` with lots of comments, TODO and bug number :)